main.tf

```diff
@@ -1,47 +1,69 @@
 resource "aws_route53_health_check" "this" {
-  fqdn                            = var.domain_name
-  port                            = var.port
-  type                            = var.type
-  resource_path                   = var.resource_path
-  failure_threshold               = var.failure_threshold
-  request_interval                = var.request_interval
-  search_string                   = var.search_string
-  measure_latency                 = var.measure_latency
-  regions                         = var.regions
-  routing_control_arn             = var.routing_control_arn
-  invert_healthcheck              = var.invert_healthcheck
+  fqdn                = var.domain_name
+  port                = var.port
+  type                = var.type
+  resource_path       = var.resource_path
+  failure_threshold   = var.failure_threshold
+  request_interval    = var.request_interval
+  search_string       = var.search_string
+  measure_latency     = var.measure_latency
+  regions             = var.regions
+  routing_control_arn = var.routing_control_arn
+  invert_healthcheck  = var.invert_healthcheck
   tags = merge(
     {
       Name = var.name
     },
   var.tags)
 }
 
-resource "aws_cloudwatch_metric_alarm" "this" {
-  alarm_name          = "${var.alarm_prefix}-${var.domain_name}${var.resource_path}"
+resource "aws_cloudwatch_metric_alarm" "alarm_breaching" {
+  alarm_name          = "${var.alarm_prefix}-${var.domain_name}${var.resource_path}-Breaching"
   namespace           = "AWS/Route53"
   metric_name         = "HealthCheckStatus"
-  comparison_operator = "LessThanThreshold"
+  comparison_operator = "LessThanOrEqualToThreshold"
   evaluation_periods  = var.evaluation_periods
   period              = var.period
-  statistic           = "Minimum"
+  statistic           = "Average"
   threshold           = var.threshold
   unit                = "None"
   dimensions = {
     HealthCheckId = aws_route53_health_check.this.id
   }
   alarm_description         = var.alarm_description == null ? "${var.domain_name}/${var.resource_path} is down" : var.alarm_description
   alarm_actions             = [aws_sns_topic.this.arn]
-  ok_actions                = [aws_sns_topic.this.arn]
   insufficient_data_actions = []
   treat_missing_data        = "breaching"
   depends_on = [
     aws_route53_health_check.this
   ]
+  tags = var.tags
+}
 
+resource "aws_cloudwatch_metric_alarm" "alarm_ok" {
```

Comment:
Create a flag to enable disable

Reply:
Also rename the resource name as just `ok` instead of `alarm_ok`

Reply:
Done